source/game_sa/CGeneral.cpp

```diff
@@ -33,7 +47,22 @@ unsigned char CGeneral::GetNodeHeadingFromVector(float x, float y) {
 
 // Converted from cdecl bool CGeneral::SolveQuadratic(float a,float b,float c,float &x1,float &x2) 0x53CE30
 bool CGeneral::SolveQuadratic(float a, float b, float c, float& x1, float& x2) {
+#ifdef USE_DEFAULT_FUNCTIONS
     return ((bool(__cdecl *)(float, float, float, float&, float&))0x53CE30)(a, b, c, x1, x2);
+#else
+    float discriminant = b * b - a * c * 4.0;
+
+    if (discriminant < 0.0) {
+        return false;
+    }
+
+    discriminant = sqrt(discriminant);
+
+    x2 = (discriminant - b) * (1.0 / a) * 0.5;
+    x1 = (-b - discriminant) * (1.0 / a) * 0.5;
+
+    return true;
```

Comment:
```suggestion
    float discriminant = b * b - a * c * 4.0f;
    if (discriminant < 0.0f) 
        return false;
    discriminant = sqrt(discriminant);
    x2 = (discriminant - b) * (1.0f / a) * 0.5f;
    x1 = (-b - discriminant) * (1.0f / a) * 0.5f;
    return true;
```
Please use the `f` prefix for floating-point values, otherwise, visual studio will spam warnings in the log. The code will work fine, but the warnings are not nice.